docs/03_goclient/15_sessions.md

```diff
@@ -0,0 +1,134 @@
+
+The returned session context should be used to execute all activities belonging to the session. The context will be cancelled if the worker executing this session dies or `CompleteSession()` is called. When using the returned session context to execute activities, a `workflow.ErrSessionFailed` error may be returned if the session framework detects that the worker executing this session has died. The failure of user activities won't affect the state of the session, so user still needs to handler the errors returned from their activites and call `CompleteSession()` if necessary.
+
+```go
+func CompleteSession(ctx Context)
+```
+
+`CompleteSession()` takes in a session context and closes it. When it's called, the session context will be canceled (means all user activities using that session context will also be canceled) and the resources reserved at the worker will be released, so make sure `CompleteSession()` is called when you no longer need the session. As it's safe to call `CompleteSession()` on a failed session, typical usage is running it as a `defer` function after a session is successfully created.
+
+### Sample Code
+
+```go
+func FileProcessingWorkflow(ctx workflow.Context, fileID string) (err error) {
+  ao := workflow.ActivityOptions{
+    ScheduleToStartTimeout: time.Second * 5,
+    StartToCloseTimeout:    time.Minute,
+  }
+  ctx = workflow.WithActivityOptions(ctx, ao)
+
+  so := &workflow.SessionOptions{
+    CreationTimeout:  time.Minute,
+    ExecutionTimeout: time.Minute,
+  }
+  sessionCtx, err := workflow.CreateSession(ctx, so)
+  if err != nil {
+    return err
+  }
+  defer workflow.CompleteSession(sessionCtx)
+
+  var fInfo *fileInfo
+  err = workflow.ExecuteActivity(sessionCtx, downloadFileActivityName, fileID).Get(sessionCtx, &fInfo)
+  if err != nil {
+    return err
+  }
+
+  var fInfoProcessed *fileInfo
+  err = workflow.ExecuteActivity(sessionCtx, processFileActivityName, *fInfo).Get(ctx, &fInfoProcessed)
+  if err != nil {
+    return err
+  }
+
+  return workflow.ExecuteActivity(sessionCtx, uploadFileActivityName, *fInfoProcessed).Get(ctx, nil)
+}
+```
+
+## Session Metadata
+
+```go
+type SessionInfo struct {
+  // A unique ID for the session
+  SessionID         string
+
+  // The hostname of the worker that is executing the session
+  HostName          string
+
+  // ... other unexported fields
+}
+
+func GetSessionInfo(ctx Context) *SessionInfo
+```
+
+The session context also stores some metadata of the session, which can be retrieved by the `GetSessionInfo()` API. For now, there are only two exported fields in the returned `workflow.SessionInfo`. We may consider exposing more information in the future.
```

Comment:
I would remove the "for now... in the future" part as we'll forget that this exists in the documentation esp if there's no issue tracking it. Just talk about the fields that are available.